tslint.json

```diff
@@ -85,6 +86,14 @@
     "jsdoc-format": [true, "check-multiline-start"],
     "no-duplicate-imports": true,
 
+    // Codelyzer
+    "banana-in-box": true,
+    // TODO: enable when https://github.com/mgechev/codelyzer/issues/545 is solved
+    "contextual-life-cycle": false,
+    "decorator-not-allowed": true,
+    "templates-no-negated-async": true,
+    "use-life-cycle-interface": true,
```

Comment:
Can you explain what these do? I'm not familiar with the specific checks.

Reply:
Of course.

**banana-in-box**: Ensure that the two-way data binding syntax is correct.
**contextual-life-cycle**: Ensure that classes use allowed life cycle method in its body, ex.:
Injectable classes should not implement or declare `ngOnInit` hook.
**decorator-not-allowed**: Ensure that classes use allowed decorator in its body. ex.:
Injectable classes should not use `@Input`.
**no-output-on-prefix**: Disallow outputs with the prefix "on".
**templates-no-negated-async**: Ensures that strict equality is used when evaluating negations on async pipe output.
**use-life-cycle-interface**: Ensure that components implement life cycle interfaces if they use them. (This is what caught the failures that are being fixed in this PR).